File: ObjectiveCExample/ObjectiveCExample/ViewController.m
#import "ViewController.h"

@implementation ViewController

- (void)promptToLoadAd {
    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Confirm"
                               message:@"Do you want to release the Ads?"
                               preferredStyle:UIAlertControllerStyleAlert];

    UIAlertAction *cancelAction = [UIAlertAction
                                   actionWithTitle:NSLocalizedString(@"Cancel", @"Cancel action")
                                   style:UIAlertActionStyleCancel
                                   handler:^(UIAlertAction *action){NSLog(@"Cancel button tapped");}];

    UIAlertAction *okAction = [UIAlertAction
                               actionWithTitle:NSLocalizedString(@"OK", @"OK action")
                               style:UIAlertActionStyleDefault
                               handler:^(UIAlertAction *action){NSLog(@"Ok button tapped"); [self loadFromConfig];}];

    [alert addAction:cancelAction];
    [alert addAction:okAction];
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UILabel *labelAbove = [[UILabel alloc]initWithFrame:CGRectMake(0, 0, 320, 100)];
    self.adView=[[AdnuntiusAdWebView alloc] initWithFrame:CGRectMake(0, 100, 320, 275)];
    UILabel *labelBelow = [[UILabel alloc]initWithFrame:CGRectMake(0, 375, 320, 125)];
    
    labelAbove.textAlignment = NSTextAlignmentCenter;
    labelAbove.backgroundColor = [UIColor brownColor];
    labelAbove.textColor = [UIColor whiteColor];
    labelAbove.text = @"Im above";
    
    labelBelow.textAlignment = NSTextAlignmentCenter;
    labelBelow.backgroundColor = [UIColor greenColor];
    labelBelow.textColor = [UIColor whiteColor];
    labelBelow.text = @"Im below";
    
    self.adView.backgroundColor = [UIColor orangeColor];
    
    self.view.backgroundColor = [UIColor blueColor];
    [self.adView setOpaque:FALSE];
    self.adView.scrollView.backgroundColor = [UIColor purpleColor];
    [self.adView.scrollView setOpaque:FALSE];
    
    [self.view setNeedsLayout];
    
    [self.view addSubview:labelAbove];
    [self.view addSubview:self.adView];
    [self.view addSubview:labelBelow];

    NSDictionary *appDefaults = [NSDictionary dictionaryWithObjectsAndKeys:
                                     @"", @"globalUserId",
                                     nil];
    [[NSUserDefaults standardUserDefaults] registerDefaults:appDefaults];
    
    NSString* globalUserId = [[NSUserDefaults standardUserDefaults] stringForKey:@"globalUserId"];
    if ([globalUserId length] == 0) {
        NSLog(@"No global user id found, generating");
        NSString *userId = [[NSUUID UUID] UUIDString];
        [[NSUserDefaults standardUserDefaults] setObject: userId forKey:@"globalUserId"];
    }
}

- (void)viewDidAppear:(BOOL)animated {
     [super viewDidAppear:animated];

    // Declare Alert message this is just so we can attach the browser debugger
    [self promptToLoadAd];
    
    //[self loadFromConfig];
}
            
- (void)loadFromConfig {
    NSString* globalUserId = [[NSUserDefaults standardUserDefaults] stringForKey:@"globalUserId"];
    NSLog(@"global user is %@", globalUserId);
    
    [self.adView enableDebug:YES];
    NSString *sessionId = [[NSUUID UUID] UUIDString];
    
    AdRequest *adRequest = [[AdRequest alloc] init:@"000000000006f450"];
    [adRequest useCookies:true];
    [adRequest userId:globalUserId];
    [adRequest sessionId:sessionId];
    [adRequest consentString:@"some consent string"];
    [adRequest keyValue:@"version" :@"6s"];
    [self.adView loadAd:adRequest :self delayViewEvents:NO];
}

- (void)onNoAdResponse:(AdnuntiusAdWebView * _Nonnull)view {
    NSLog(@"No ad found");
    self.adView.hidden = true;
}

- (void)onFailure:(AdnuntiusAdWebView * _Nonnull)view :(NSString * _Nonnull)message {
    NSLog(@"Failure: %@", message);
    self.adView.hidden = true;
}

- (void)onAdResponse:(AdnuntiusAdWebView * _Nonnull)view :(AdResponseInfo * _Nonnull)response {
    NSLog(@"ad found, height: %1ld, width: %1ld", response.definedHeight, response.definedWidth);
    
    if (response.definedHeight > 0) {
        CGRect frame = view.frame;
        frame.size.height = response.definedHeight;
        view.frame = frame;
    }
}

@end

File: SwiftLayoutExample/SwiftLayoutExample/ViewController.swift
//
//  ViewController.swift
//  Example3
//
//  Copyright © 2019 Adnuntius. All rights reserved.
//

import WebKit
import AdnuntiusSDK

class ViewController: UIViewController, LoadAdHandler {
    @IBOutlet weak var adView: AdnuntiusAdWebView!
    
    fileprivate func promptToLoadAd() {
        let dialogMessage = UIAlertController(title: "Confirm", message: "Do you want to release the Ads?", preferredStyle: .alert)
        
        let ok = UIAlertAction(title: "OK", style: .default, handler: { (action) -> Void in
            print("Ok button tapped")
            self.loadFromConfig()
        })
        
        let cancel = UIAlertAction(title: "Cancel", style: .cancel) { (action) -> Void in
            print("Cancel button tapped")
        }
        
        dialogMessage.addAction(ok)
        dialogMessage.addAction(cancel)
        
        self.present(dialogMessage, animated: true, completion: nil)
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        // Declare Alert message this is just so we can attach the browser debugger
        promptToLoadAd()

        //self.loadFromConfig()
    }
    
    private func loadFromConfig() {
        adView.logger.debug = true
        
        let adRequest = AdRequest("000000000006f450")
        adRequest.keyValue("version", "6s")
        adView.loadAd(adRequest, self, delayViewEvents: false)
    }
    
    func onNoAdResponse(_ view: AdnuntiusAdWebView) {
        print("No Ad Found!")
        view.isHidden = true
    }
    
    func onFailure(_ view: AdnuntiusAdWebView, _ message: String) {
        view.loadHTMLString("<h1>Error is: \(message)</h1>", baseURL: nil)
    }
    
    func onAdResponse(_ view: AdnuntiusAdWebView, _ response: AdResponseInfo) {
        print("onAdResponse: width: \(response.definedWidth), height: \(response.definedHeight)")
        var frame = view.frame
        if (response.definedHeight > 0) {
            frame.size.height = CGFloat(response.definedHeight)
        }
        view.frame = frame
    }
}
